.github/workflows/build_and_upload.yml

```diff
@@ -0,0 +1,67 @@
+name: Build and upload to PyPI
+
+on:
+  push:
+  pull_request:
+  release:
+    types:
+      - published
+
+jobs:
+  build_wheels:
+    name: "Build wheels on ${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - uses: "actions/checkout@v3"
+        with:
+          submodules: true
+      - name: "Build wheels"
+        uses: "pypa/cibuildwheel@v2.3.1"
+        env:
+          CIBW_SKIP: "pp38-win_amd64"  # FIXME
+          CIBW_BEFORE_BUILD: "pip install -U cython && ./update_cpp.sh"
+          CIBW_BEFORE_BUILD_WINDOWS: "pip install -U cython && update_cpp.sh"
+          CIBW_TEST_COMMAND: "echo Wheel installed successfully"
```

Comment:
can you try:

CIBW_TEST_REQUIRES=tox
CIBW_TEST_COMMAND="cd {project} && tox"

Reply:
> @Gerrit-K, looks like we are close. if we can figure out how to get tox to only run the relevant python version, then I think this can come in.

Yes, was about to comment that, but then left for dinner ;)
 
The question is whether we really need tox now, after all. As far as I understand it, it is designed to perform isolated testing across several python versions. Now, since cibuildwheel already provides isolated environments for the built wheels, couldn't we just simply run pytest instead of tox? Or is there more to it that I currently dont understand (yet)?

Reply:
i think directly calling pytest is good, let’s do that

Reply:
Alright, getting closer. All tests until PyPy 3.7 x64 succeed. That one fails on all OSes with `SystemError: Function returned a NULL result without setting an exception`. Any ideas?

Reply:
let's skip the failing pypy builds.